lib/internal/crypto/sig.js

```diff
@@ -13,10 +13,13 @@ const {
   getDefaultEncoding,
   toBuf
 } = require('internal/crypto/util');
+const { isArrayBufferView } = require('internal/util/types');
 const { Writable } = require('stream');
 const { inherits } = require('util');
 
 function Sign(algorithm, options) {
+  if (typeof algorithm !== 'string')
```

Comment:
Nit: maybe it's better to move this after the `instanceof` check to only run it once when `Sign` is called without `new`.